### ISSUE TITLE
running temporary agent disconnects client desktop (sometimes whole connection)

### ISSUE THREAD

**Issue by Ridd01** [Ylianst/MeshCentral#2209]
Like in title, when I'm runing "temporary agent" on client computer, after connecting to desktop session and moving windows (doing something on screen), agent disconnects desktop session (about 10-30 sec of work). I can connect right away again. This thing dont happen when Im not moving windows. Client can move windows and session stays on. Sometimes session disconnects permanently (agent window on client computer stays on and "connected").
This does not happens on installed "background agents" and same computers.
Im using apache2 with reverse proxy (behind router, forwarded ports 80 and 443).
Client systems - windows 10.

**Comment by matt-cylanda** [comment]
I've also been having this happen. I haven't done a temporary agent in a week or so, but has definitely been an annoying issue.

**Comment by LPJon** [comment]
So have I....makes it almost unusable.

**Comment by krayon007** [comment]
I was able to reproduce this issue when running in temporary mode, so am looking into it now...

**Comment by krayon007** [comment]
Ok, I found the problem and fixed it... It will be in the next agent update.
The issue was on Temp Mode agents, the KVM runs in-proc to the agent (normally it runs out-of-proc). Anyways, there was a bug, where a remote-input packet could cause the KVM to corrupt the TLS frame when running in this mode, which caused the websocket to drop.

**Comment by Ylianst** [comment]
Nice! I will be running the latest Mesh Agent thru testing first before including it in an upcoming MeshCentral release.

**Comment by JohnyHHH** [comment]
I faced the similar issue with temporary agent too.
Is this fix already include in new version? If yes, may I know which version as I would like to test on this too.

**Comment by krayon007** [comment]
> I faced the similar issue with temporary agent too.
> Is this fix already include in new version? If yes, may I know which version as I would like to test on this too.

Not yet, I'll post back when Ylian pushed the update

**Comment by JohnyHHH** [comment]
> > I faced the similar issue with temporary agent too.
> > Is this fix already include in new version? If yes, may I know which version as I would like to test on this too.
> 
> Not yet, I'll post back when Ylian pushed the update

Alright! thanks you.

**Comment by JohnyHHH** [comment]
Hi guys, would like to check is this updates already in v0.7.68? 
I tried with v0.7.68, the temporary agent does not disconnect, however after 10minutes, the UI freeze during remote desktop.
But the mouse and keyboard input even is still works when I try to move my mouse and type something.

**Comment by krayon007** [comment]
Yes the update is applied to the current version... The issue you are currently seeing is likely unrelated, but I'll see if I can reproduce that.

**Comment by JohnyHHH** [comment]
> Yes the update is applied to the current version... The issue you are currently seeing is likely unrelated, but I'll see if I can reproduce that.

Thanks.

**Comment by JohnyHHH** [comment]
Hi @krayon007 ,

I was halfway testing on the temporary agent.
The agent was disconnected and this shown
![image](https://user-images.githubusercontent.com/58418444/108144596-fb9b5100-7104-11eb-8343-3182df2facdb.png)

Got this in the log
[2021-02-17 09:18:04 AM] Info: No certificate was found in db
[2021-02-17 09:43:02 AM] ..\microstack\ILibParsers.c:1091 (254,0) (null)

This is my first time encountered this, I'm not sure if this have anything related to the freeze issue or not? 
The freeze issue seen previously does not show any error and the agent was not being disconnected.

**Comment by JohnyHHH** [comment]
Hi @krayon007 , Is there log file that we can looks at what happen when it freeze?
Or a refresh function that can refresh the fetch the screen updates? 
I notice that when it freeze, try to disconnect and connect again, the remote desktop just who black screen.

**Comment by JohnyHHH** [comment]
I notice that the freeze happened the agent PC is used by user.
While remote desktop to the agent PC, without physically using the agent PC (as to perform keyboard/mouse actions on it). My remote desktop session is working fine.
When someone use the remote agent PC while remote session is running, this freeze issue will happened.

**Comment by Ridd01** [comment]
> 
> 
> I notice that the freeze happened the agent PC is used by user.
> While remote desktop to the agent PC, without physically using the agent PC (as to perform keyboard/mouse actions on it). Remote desktop session did not show freeze issue.
> When someone use the remote agent PC while remote session is running, this freeze issue will happened.

I can confirm that

**Comment by JohnyHHH** [comment]
Hi @krayon007 , do you able to reproduce this?

**Comment by krayon007** [comment]
> Hi @krayon007 , do you able to reproduce this?

Not yet. I left it connected for around an hour, and didn't have any issues.

**Comment by Ridd01** [comment]
> 
> 
> > Hi @krayon007 , do you able to reproduce this?
> 
> Not yet. I left it connected for around an hour, and didn't have any issues.

Try use computer while connected and other person should try to remotely move simultaneously mouse and try to click things with you.

**Comment by JohnyHHH** [comment]
Hi @krayon007 , Any chance this can be fix?

**Comment by yomono** [comment]
We are still facing this issue. In temporary mode, the agent disconnects constantly, until we just can't reconnect again.
This is unusable to the point we are asking the customers to simply download AnyDesk.
We don't want to use Anydesk, we want meshcentral!!

**Comment by Ridd01** [comment]
> 
> 
> We are still facing this issue. In temporary mode, the agent disconnects constantly, until we just can't reconnect again.
> This is unusable to the point we are asking the customers to simply download AnyDesk.
> We don't want to use Anydesk, we want meshcentral!!

it's obvious we want to use MeshCentral - cus we love it! Please, don't mention AnyDesk here - cus we dontt love it (?) ... ;)
And yes, it still shows up periodically (the problem)...
